daml-lf/kv-support/src/main/scala/com/daml/lf/kv/archives/ArchiveConversions.scala

```diff
@@ -3,31 +3,49 @@
 
 package com.daml.lf.kv.archives
 
-import com.daml.lf.archive.ArchiveParser
+import com.daml.lf.archive.{ArchiveParser, Decode, Error => ArchiveError}
 import com.daml.lf.data.Ref
-import com.daml.lf.kv.ConversionError
+import com.daml.lf.data.Ref.PackageId
+import com.daml.lf.language.Ast
 
 object ArchiveConversions {
 
-  def parsePackageId(rawArchive: RawArchive): Either[ConversionError.ParseError, Ref.PackageId] =
+  def parsePackageId(rawArchive: RawArchive): Either[ArchiveError, Ref.PackageId] =
     for {
-      archive <- ArchiveParser
-        .fromByteString(rawArchive.byteString)
-        .left
-        .map(error => ConversionError.ParseError(error.msg))
+      archive <- ArchiveParser.fromByteString(rawArchive.byteString)
       packageId <- Ref.PackageId
         .fromString(archive.getHash)
         .left
-        .map(ConversionError.ParseError)
+        .map(ArchiveError.Parsing)
     } yield packageId
 
   def parsePackageIdsAndRawArchives(
       archives: List[com.daml.daml_lf_dev.DamlLf.Archive]
-  ): Either[ConversionError.ParseError, Map[Ref.PackageId, RawArchive]] =
+  ): Either[ArchiveError.Parsing, Map[Ref.PackageId, RawArchive]] =
     archives.partitionMap { archive =>
       Ref.PackageId.fromString(archive.getHash).map(_ -> RawArchive(archive.toByteString))
     } match {
       case (Nil, hashesAndRawArchives) => Right(hashesAndRawArchives.toMap)
-      case (errors, _) => Left(ConversionError.ParseError(errors.head))
+      case (errors, _) => Left(ArchiveError.Parsing(errors.head))
     }
+
+  def decodePackages(
+      hashesAndArchives: Iterable[RawArchive]
+  ): Either[ArchiveError, Map[Ref.PackageId, Ast.Package]] = {
+    type Result = Either[ArchiveError, Map[Ref.PackageId, Ast.Package]]
```

Comment:
What benefits do we get from this type in this limited scope?

Reply:
The line is shorter. I just moved it one-to-one. Can inline it if you'd like.

Reply:
Fine either way for me, I was just expecting to have the type used in more places.